clang 4.0.1 build failure on -Waddress-of-packed-member

```
==> ./waf configure --prefix=/usr/local/Cellar/serialosc/HEAD-2aba0b7
Setting top to                           : /private/tmp/serialosc-20170824-61193-1c1eco9 
Setting out to                           : /private/tmp/serialosc-20170824-61193-1c1eco9/build 

Checking for 'clang' (C compiler)        : clang 

Checking for working poll()              : no (will use select()) 
Checking for libmonome                   : yes 
Checking for liblo                       : yes 
Checking for function strdup             : yes 
Checking for function _strdup            : not found 
Checking for function strndup            : yes 
Checking for function strcasecmp         : yes 
Checking for header unistd.h             : yes 

'configure' finished successfully (1.418s)
==> ./waf build
Waf: Entering directory `/private/tmp/serialosc-20170824-61193-1c1eco9/build'
[ 1/51] Compiling third-party/libuv/src/fs-poll.c
[ 2/51] Compiling third-party/libuv/src/inet.c
[ 3/51] Compiling third-party/libuv/src/threadpool.c
[ 4/51] Compiling third-party/libuv/src/uv-common.c
[ 5/51] Compiling third-party/libuv/src/version.c
[ 6/51] Compiling third-party/libuv/src/unix/async.c
[ 7/51] Compiling third-party/libuv/src/unix/core.c
[ 8/51] Compiling third-party/libuv/src/unix/dl.c
[ 9/51] Compiling third-party/libuv/src/unix/fs.c
[10/51] Compiling third-party/libuv/src/unix/getaddrinfo.c
[11/51] Compiling third-party/libuv/src/unix/getnameinfo.c
[12/51] Compiling third-party/libuv/src/unix/loop.c
[13/51] Compiling third-party/libuv/src/unix/loop-watcher.c
[14/51] Compiling third-party/libuv/src/unix/pipe.c
[15/51] Compiling third-party/libuv/src/unix/poll.c
[16/51] Compiling third-party/libuv/src/unix/process.c
[17/51] Compiling third-party/libuv/src/unix/signal.c
[18/51] Compiling third-party/libuv/src/unix/stream.c
[19/51] Compiling third-party/libuv/src/unix/tcp.c
[20/51] Compiling third-party/libuv/src/unix/thread.c
[21/51] Compiling third-party/libuv/src/unix/timer.c
[22/51] Compiling third-party/libuv/src/unix/tty.c
[23/51] Compiling third-party/libuv/src/unix/udp.c
[24/51] Compiling third-party/libuv/src/unix/proctitle.c
[25/51] Compiling third-party/libuv/src/unix/darwin.c
[26/51] Compiling third-party/libuv/src/unix/fsevents.c
[27/51] Compiling third-party/libuv/src/unix/darwin-proctitle.c
[28/51] Compiling third-party/libuv/src/unix/kqueue.c
[29/51] Compiling third-party/confuse/confuse.c
[30/51] Compiling third-party/confuse/lexer.c
[31/51] Compiling src/common/platform/posix.c
[32/51] Compiling src/common/platform/darwin.c
[33/51] Compiling src/common/ipc.c
[34/51] Compiling src/common/util.c
[35/51] Compiling src/serialoscd/uv.c
[36/51] Compiling src/serialosc-detector/iokitlib.c
../src/common/ipc.c:149:28: error: taking address of packed member 'devnode' of class or structure 'sosc_ipc_msg::(anonymous union)::(anonymous)' may result in an unaligned pointer value [-Werror,-Waddress-of-packed-member]
                if (read_strdata(fd, 1, &buf->connection.devnode))
                                         ^~~~~~~~~~~~~~~~~~~~~~~
../src/common/ipc.c:157:28: error: taking address of packed member 'serial' of class or structure 'sosc_ipc_msg::(anonymous union)::(anonymous)' may result in an unaligned pointer value [-Werror,-Waddress-of-packed-member]
                if (read_strdata(fd, 2, &buf->device_info.serial,
                                         ^~~~~~~~~~~~~~~~~~~~~~~
../src/common/ipc.c:158:21: error: taking address of packed member 'friendly' of class or structure 'sosc_ipc_msg::(anonymous union)::(anonymous)' may result in an unaligned pointer value [-Werror,-Waddress-of-packed-member]
                                 &buf->device_info.friendly))
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~
3 errors generated.

Waf: Leaving directory `/private/tmp/serialosc-20170824-61193-1c1eco9/build'
Build failed
 -> task in 'serialosc-common' failed (exit status 1): 
	{task 4507184016: c ipc.c -> ipc.c.2.o}
['clang', '-mmacosx-version-min=10.5', '-std=c99', '-Wall', '-Werror', '-I/private/tmp/serialosc-20170824-61193-1c1eco9/build', '-I/private/tmp/serialosc-20170824-61193-1c1eco9', '-I/private/tmp/serialosc-20170824-61193-1c1eco9/build/include', '-I/private/tmp/serialosc-20170824-61193-1c1eco9/include', '-I/private/tmp/serialosc-20170824-61193-1c1eco9/build/third-party', '-I/private/tmp/serialosc-20170824-61193-1c1eco9/third-party', '-DHAVE_LIBMONOME=1', '-DHAVE_LO=1', '-DHAVE_STRDUP=1', '-DHAVE_STRNDUP=1', '-DHAVE_STRCASECMP=1', '-DHAVE_UNISTD_H=1', '-DVERSION="1.4"', '-D_GNU_SOURCE=1', '-DGIT_COMMIT="2aba0b7"', '../src/common/ipc.c', '-c', '-o', '/private/tmp/serialosc-20170824-61193-1c1eco9/build/src/common/ipc.c.2.o']
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python.rb
/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
BuildError: Failed executing: ./waf build
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 5
When you exit this shell, you will return to the menu.
bash-3.2$ clang --version
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
bash-3.2$
```

This causes the build to fail on macOS Sierra and above with >= Xcode 8.3. It will also fail with any version of LLVM 4's clang regardless of platform.

CC @wrl

--- a/include/serialosc/ipc.h
+++ b/include/serialosc/ipc.h
@@ -18,14 +18,12 @@
 
 #include <stdint.h>
 
-#ifndef PACKED
-#define PACKED __attribute__((__packed__))
-#endif
-
 #ifdef WIN32
 #define SOSC_PIPE_PREFIX "\\\\.\\pipe\\org.monome.serialosc-"
 #define SOSC_DETECTOR_PIPE (SOSC_PIPE_PREFIX "detector")
 #endif
+
+#define SOSC_IPC_MSG_BUFFER_SIZE 256
 
 typedef enum {
 	/* device -> supervisor */
@@ -45,20 +43,20 @@ typedef struct sosc_ipc_msg {
 	__extension__ union {
 		struct {
 			char *devnode;
-		} PACKED connection;
+		} connection;
 
 		struct {
 			char *serial;
 			char *friendly;
-		} PACKED device_info;
+		} device_info;
 
 		struct {
 			uint16_t port;
-		} PACKED port_change;
+		} port_change;
 	};
 
 	uint16_t magic;
-} PACKED sosc_ipc_msg_t;
+} sosc_ipc_msg_t;
 
 int sosc_ipc_msg_write(int fd, sosc_ipc_msg_t *msg);
 int sosc_ipc_msg_read(int fd, sosc_ipc_msg_t *buf);

--- a/src/common/ipc.c
+++ b/src/common/ipc.c
@@ -75,7 +75,7 @@ read_strdata(int fd, size_t n, ...)
 int
 sosc_ipc_msg_write(int fd, sosc_ipc_msg_t *msg)
 {
-	uint8_t buf[64];
+	uint8_t buf[SOSC_IPC_MSG_BUFFER_SIZE];
 	ssize_t written;
 	ssize_t bufsiz;
 

--- a/src/serialosc-detector/windows.c
+++ b/src/serialosc-detector/windows.c
@@ -42,7 +42,7 @@ struct detector_state {
 static void
 send_connect(const struct detector_state *state, char *port)
 {
-	uint8_t buf[64];
+	uint8_t buf[SOSC_IPC_MSG_BUFFER_SIZE];
 	DWORD written;
 	size_t bufsiz;
 

--- a/src/serialosc-device/server.c
+++ b/src/serialosc-device/server.c
@@ -163,7 +163,7 @@ static void
 send_ipc_msg(sosc_ipc_msg_t *msg)
 {
 	HANDLE p = (HANDLE) _get_osfhandle(STDOUT_FILENO);
-	uint8_t buf[64];
+	uint8_t buf[SOSC_IPC_MSG_BUFFER_SIZE];
 	DWORD written;
 	ssize_t bufsiz;
 

--- a/src/serialoscd/uv.c
+++ b/src/serialoscd/uv.c
@@ -246,7 +246,7 @@ write_ipc_msg_to_stream(uv_stream_t *stream, struct sosc_ipc_msg *msg)
 {
 	uv_buf_t uv_buf;
 	uv_write_t *req;
-	uint8_t buf[64];
+	uint8_t buf[SOSC_IPC_MSG_BUFFER_SIZE];
 	ssize_t nbytes;
 
 	nbytes = sosc_ipc_msg_to_buf(buf, sizeof(buf), msg);

